### PR TITLE
implement CAI mint/burn

### PR DESCRIPTION
**Pull request by 0xlock**

### CODE REVIEW

File: contracts/facets/avalanche/CaiFacet.sol [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+    address private constant CAI = 0x48f88A3fE843ccb0b5003e70B4192c1d7448bEf0;
+
+    function mintCai(
+        bytes4 selector,
+        bytes memory data,
+        address fromToken,
+        uint256 fromAmount,
+        uint256 minOut
+    ) external nonReentrant onlyOwner recalculateAssetsExposure remainsSolvent {
```
Comment (kavsky02) [inline contracts/facets/avalanche/CaiFacet.sol:30]: Let's not use `recalculateAssetsExposure` (we're removing it now) and instead use `SolvencyMethods::_decreaseExposure` and `SolvencyMethods::_increaseExposure` that handle exposure + adding/removing owned assests

Comment (kavsky02) [reply contracts/facets/avalanche/CaiFacet.sol:30]: seems like the `recalculateAssetsExposure` modifier is still there @0xlock

**Comment by kavsky02** [comment]
Added `onlyOwnerOrInsolvent` instead of `onlyOwner` to `burnCai()`

File: contracts/facets/avalanche/CaiFacet.sol [hunk 2]
```diff
@@ -109,6 +121,29 @@ contract CaiFacet is ReentrancyGuardKeccak, OnlyOwnerOrInsolvent {
...
+    ) internal {
+        bytes32[] memory assets = new bytes32[](2);
+        assets[0] = inputAsset;
+        assets[1] = outputAsset;
+
+        uint256[] memory prices = getOracleNumericValuesFromTxMsg(assets);
+
+        uint256 inputValue = prices[0] * inputAmount;
+        uint256 outputValue = prices[1] * outputAmount;
+        uint256 diff = inputValue > outputValue
```
Comment (mbare0) [inline contracts/facets/avalanche/CaiFacet.sol:139]: If `inputValue` < `outputValue`, we should accept such slippage (user just profits from a tx like that)

**Pull request merged**